pkg/cloudprovider/aws/launchtemplate.go

```
@@ -349,9 +350,20 @@ exec > >(tee /var/log/user-data.log|logger -t user-data -s 2>/dev/console) 2>&1
 		userData.WriteString(fmt.Sprintf(` \
     --kubelet-extra-args '%s'`, kubeletExtraArgs))
 	}
-	if len(constraints.KubeletConfiguration.ClusterDNS) > 0 {
+	if dnsClusterIp := constraints.KubeletConfiguration.ClusterDNS; len(dnsClusterIp) > 0 {
 		userData.WriteString(fmt.Sprintf(` \
-    --dns-cluster-ip '%s'`, constraints.KubeletConfiguration.ClusterDNS[0]))
+    --dns-cluster-ip '%s'`, dnsClusterIp[0]))
+
+		ipAddr, ipNet, _ := net.ParseCIDR(dnsClusterIp[0] + "/108")
```

We should panic if we fail to parse instead of squashing the error with `_`.